File: python/day1/collect_measurements.py
from collections.abc import Sequence


def collect_measurements(arr: Sequence[str]) -> int:
    if not arr or len(arr) == 0:
        return 0

    increased = 0

    for i in range(0, len(arr)):
        if arr[i] > arr[i-1]:
            increased = increased + 1
    return increased


def collect_measurements_sliding_window(arr: Sequence[str]) -> int:
    if not arr or len(arr) == 0:
        return 0

    def sum(array, index): return array[index] + \
        array[index + 1] + array[index + 2]
    
    increased = 0

    for i in range(0, len(arr)):
        if len(arr) > i + 3 and sum(arr, i + 1) > sum(arr, i):
            increased = increased + 1
    
    return increased

File: javascript/day1/collect_measurements.js
function collectMeasurements(arr) {
  if (!arr || arr.length === 0) {
    return 0;
  }

  let increased = 0;

  for (let i = 0; i < arr.length; i++) {
    if (arr[i] >= arr[i - 1]) {
      increased = increased + 1;
    }
  }

  return increased;
}

function collectMeasurementsSlidingWindow(arr) {
  if (!arr || arr.length === 0) {
    return 0;
  }

  const sum = (array, index) =>
    array[index] + array[index + 1] + array[index + 2];

  let increased = 0;

  for (let i = 0; i < arr.length; i++) {
    if (arr[i + 2] && sum(arr, i + 1) > sum(arr, i)) {
      increased++;
    }
  }

  return increased;
}

module.exports = {
  collectMeasurements,
  collectMeasurementsSlidingWindow,
};
